tests/test_mesos.py

```diff
@@ -0,0 +1,59 @@
+# stdlib imports
+import re
+
+# third-party imports
+import responses
+
+# local imports
+from shpkpr.mesos import resolve_leader_url
+from shpkpr.mesos import MesosClient
+
+
+def _mock_valid_redirect():
+    responses.add(responses.HEAD,
+                  'http://master.example.com:5050/master/redirect',
+                  status=307,
+                  adding_headers={'Location': '//leader.example.com:5050'})
+    responses.add(responses.HEAD,
+                  re.compile(r'http://leader\.example\.com:5050/?'),
+                  status=200)
+
+
+@responses.activate
+def test_resolve_leader_valid_redirect():
+    _mock_valid_redirect()
+
+    leader_url = resolve_leader_url('http://master.example.com:5050')
+    assert leader_url == "http://leader.example.com:5050"
+
+
+@responses.activate
+def test_resolve_leader_valid_redirect_with_trailing_slash():
+    _mock_valid_redirect()
+
+    leader_url = resolve_leader_url('http://master.example.com:5050/')
+    assert leader_url == "http://leader.example.com:5050"
+
+
+@responses.activate
+def test_mesos_client_resolves_leader_url():
+    _mock_valid_redirect()
+
+    client = MesosClient('http://master.example.com:5050')
+    # testing private members of a class isn't usually encouraged, however, in
+    # this case the private members under test are the parts we've overridden
+    # from dcos.mesos.DCOSClient and we need to test our implementation.
+    assert client._leader_url is None
+    assert client._mesos_master_url == "http://leader.example.com:5050"
+    assert client._leader_url == "http://leader.example.com:5050"
```

Comment:
Something feels a bit strange here. Seems like we're asserting that the call to `_mesos_master_url` changes the property `_leader_url` ... is there any way to ensure that `_leader_url` is a read only property? Also is there any construct to test that a value is changing in this test rather than asserting 3 times? It's a bit tough to tell what the test is trying to achieve.